pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -1070,6 +1071,21 @@ func (os *OpenStack) CreateVolume(name string, size int, tags *map[string]string
 	return vol.ID, err
 }
 
+func (os *OpenStack) GetDevicePath(diskId string) string {
+	files, _ := ioutil.ReadDir("/dev/disk/by-id/")
```

Comment:
The usage of the specific device path `/dev/disk/by-id/virtio-<UUID>` makes a couple of assumptions:

1. That the device in question is from a pure/upstream OpenStack install, and not a Rackspace Public Cloud block device, as Rackspace presents block devices under the path schema `/dev/disk/by-uuid/<UUID>`
2. And that the block device in question was formatted with a fs type after its creation
   
    * By default a new block device in OpenStack or Rackspace comes clean **without** a fs on it where the path is `/dev/vd*` on OpenStack and `/dev/xvd*` on Rackspace.
    * Not having a fs on the device at creation, but rather after a user specifically performs `mkfs.ext*` on it, means that the device will not exist in `/dev/disk/by-uuid/<UUID>` until _after_ its been formatted with a fs

I confirmed this today on fresh installs of Ubuntu 15.10 VM, as well as a CoreOS 835.9.0 VM, on the Rackspace Public Cloud.

`GetDevicePath` would have to be altered to accommodate for this difference in environment.

Also, see: https://ask.openstack.org/en/question/50882/are-devdiskby-id-symlinks-unreliable/